docs/reference/rest-api/security/query-user.asciidoc

```diff
@@ -0,0 +1,309 @@
+
+NOTE: As opposed to the <<security-api-get-user,get user api>>, <<built-in-users,built-in users>> are excluded from the
+result. This API is only for <<native-realm,native users>>.
+
+[[security-api-query-user-request]]
+==== {api-request-title}
+
+`GET /_security/_query/user`
+
+`POST /_security/_query/user`
+
+[[security-api-query-user-prereqs]]
+==== {api-prereq-title}
+
+* To use this API, you must have at least the  `read_security` cluster privileges.
+
+[[security-api-query-user-desc]]
+==== {api-description-title}
+
+Use this API to retrieve users managed by the
+<<native-realm,native realm>> in a paginated manner.
+You can optionally filter the results with a query.
+
+[[security-api-query-user-request-body]]
+==== {api-request-body-title}
+
+You can specify the following parameters in the request body:
+
+`query`::
+(Optional, string) A <<query-dsl,query>> to filter which users to return.
+The query supports a subset of query types, including
+<<query-dsl-match-all-query,`match_all`>>, <<query-dsl-bool-query,`bool`>>,
+<<query-dsl-term-query,`term`>>, <<query-dsl-terms-query,`terms`>>,
+<<query-dsl-prefix-query,`prefix`>>, <<query-dsl-wildcard-query,`wildcard`>> and <<query-dsl-exists-query,`exists`>>.
++
+You can query the following public values associated with a user.
++
+.Valid values for `query`
+[%collapsible%open]
+====
+`username`::
+An identifier for the user.
+
+`roles`::
+An array of the role names of the <<defining-roles, roles>> that are assigned to the user.
+
+`full_name`::
+Full name of the user.
+
+`email`::
+The email of the user.
+
+`enabled`::
+Specifies whether the user is enabled.
+
+====
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=from]
++
+By default, you cannot page through more than 10,000 hits using the `from` and
+`size` parameters. To page through more hits, use the
+<<search-after,`search_after`>> parameter.
+
+`size`::
+(Optional, integer) The number of hits to return. Must not be negative and defaults to `10`.
++
+By default, you cannot page through more than 10,000 hits using the `from` and
+`size` parameters. To page through more hits, use the
+<<search-after,`search_after`>> parameter.
+
+`sort`::
+(Optional, object) <<sort-search-results,Sort definition>>. You can sort on `username`, `roles` or `enabled`.
+In addition, sort can also be applied to the `_doc` field to sort by index order.
+
+`search_after`::
+(Optional, array) <<search-after,Search after>> definition.
+
+
+[[security-api-query-user-response-body]]
+==== {api-response-body-title}
+
+This API returns the following top level fields:
+
+`total`::
+The total number of users found.
+
+`count`::
+The number of users returned in the response.
+
+`users`::
+A list of users that match the query.
+
+[[security-api-query-user-example]]
+==== {api-examples-title}
+
+The following request lists all users, assuming you have the
+`read_security` privilege:
+
+[source,console]
+----
+GET /_security/_query/user
```

Comment:
I did not dig into the CI failure, but it's likely related to the missing test setup here. Doc tests are pretty finicky but could be worth working on here. 

It's possible to set up actual users under the hood by adding setup calls (grep  for `// TEST[setup:jacknich_user]` in  `get-users.asciidoc` and the test setup for it (https://github.com/elastic/elasticsearch/blob/4f4e6133e1e29af1945f066565d62435f3a6bccf/docs/build.gradle#L1770).

We could do this for good old jacknich and one more user. Then we can assert on the actual results throughout the docs file (using `get-users` as a reference for how that works). 

Would you be up for adding that?

Reply:
Thanks! Took some time to figure this out. Looks like it's working now. Ended up not adding it to the more complex query since it requires 5 users.